netshare/generate/denormalize_fields.py

```diff
@@ -32,14 +41,66 @@ def _denormalize_by_fields_list(
     """
     denormalized_data = []
     dim = 0
+    word2vec_config = get_config(
+        ["pre_post_processor.config", "learn.word2vec"], default_value={}
+    )
+    if not word2vec_config:
+        logger.debug("No learn.word2vec config found, skipping the embedding model")
+
+    session_key_fields = word2vec_config.get(
+        "session_key", word2vec_config.get("metadata", [])
+    )
+    word2vec_cols = [
+        m
+        for m in (session_key_fields + word2vec_config.get("timeseries", []))
+        if "word2vec" in m.get("encoding", "")
+    ]
+    word2vec_cols_names = [
+        m["column"]
+        for m in (session_key_fields + word2vec_config.get("timeseries", []))
+        if "word2vec" in m.get("encoding", "")
+    ]
+    word2vec_model_path = os.path.join(
+        learn_api.get_word2vec_model_directory(),
+        "{}_{}.model".format(
+            word2vec_config["word2vec"]["model_name"],
+            word2vec_config["word2vec"]["vec_size"],
+        ),
+    )
+
+    canonical_data_dir = get_canonical_data_dir()
+    df, _ = load_dataframe_chunks(canonical_data_dir)
+    dict_type_annDictPair = build_annoy_dictionary_word2vec(
+        df=df,
+        model_path=word2vec_model_path,
+        word2vec_cols=word2vec_cols,
+        word2vec_size=word2vec_config["word2vec"]["vec_size"],
+        n_trees=1000,
+    )
+
+    word2vec_dict_type_cols = get_word2vec_type_col(word2vec_cols)
+
     for field in fields_list:
         if is_session_key:
             sub_data = normalized_data[:, dim : dim + field.getOutputDim()]
         else:
             sub_data = normalized_data[:, :, dim : dim + field.getOutputDim()]
+
         sub_data = field.denormalize(sub_data)
-        if isinstance(field, ContinuousField):
-            sub_data = sub_data[:, 0] if is_session_key else sub_data[:, :, 0]
+        if field.name in word2vec_cols_names:
+            word2vec_type_col = None
+            for k in word2vec_dict_type_cols.keys():
```

Comment:
```suggestion
            for k in word2vec_dict_type_cols:
```
Just a syntax sugar :-)